test/integration/targets/task_resolved_fqcn_action/callback_plugins/display_resolved_action.py

```diff
@@ -0,0 +1,78 @@
+# (c) 2020 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = '''
+    name: display_resolved_action
+    type: stdout
+    short_description: Inherits from the default callback but displays resolved action names.
+    description:
+        - Inherits from the default callback but displays resolved action names.
+    extends_documentation_fragment:
+      - default_callback
+    requirements:
+      - set as stdout in configuration
+'''
+
+from ansible import constants as C
+from ansible.plugins.callback.default import CallbackModule as CallbackBase
+
+
+class CallbackModule(CallbackBase):
+    CALLBACK_VERSION = 2.0
+    CALLBACK_TYPE = 'stdout'
+    CALLBACK_NAME = 'display_resolved_action'
+
+    def _task_start(self, task, prefix=None):
```

Comment:
Rather than replacing/augmenting the default display plugin (with fragile internal overrides), this callback could just be a secondary callback plugin that dumps the task attributes `runme.sh` is looking for directly to stdout. It could also implement more sopshisticated test logic internally, eg, store each task.action/task.resolved_action pair in a global in `v2_playbook_on_task_start`, then in `v2_playbook_on_stats` compare them to what was expected from the playbook, and output a forensic listing and a single easy to grep string like `***PASS***` to stdout- then you only have to look for one thing in `runme.sh` without relying on internal details of the `default` callback that might change.

Reply:
Okay, that makes sense :+1: Will fix.